lib/Transforms/Scalar/DxilLoopUnroll.cpp

```diff
@@ -673,6 +680,44 @@ bool DxilLoopUnroll::IsLoopSafeToClone(Loop *L) {
   return true;
 }
 
+void DxilLoopUnroll::RecursivelyCreateSubLoopForIteration(LPPassManager &LPM, LoopInfo *LI, Loop *OuterL, Loop *L, LoopIteration &Iter) {
```

Comment:
As I read this, it replaces the loop that previously contained the subloop `L` with a new dummy loop that just contains `L` and no other subloops. Is that correct? I don't think the name of the function is particularly helpful to understanding that. Calling it something like `ReplaceParentWithDummy` or adding a comment might clarify things